### DISCUSSION TITLE
Single bar - two monitors

### DISCUSSION THREAD

**Discussion by PhrantiK** [FelixKratz/SketchyBar#213] [Q&A] [answered]
I have a main monitor in front of me and a side monitor at 90 degrees. Only the main screen has the bar displayed. 

Main screen has spaces 1-3 and side screen has spaces 4-6. 

Is there a way to make an icon highlight the space that has focus on the side monitor?

For example I have 6 icons on the bar on monitor one. If I switch to space 4 which is on the side monitor, I want only the 4th icon to highlight. Currently only 1-3 change color.

**Comment by FelixKratz** [comment]
Yes, try to set `ignore_association=on` for the space items that you want to show on all displays.

**Comment by PhrantiK** [comment]
Thanks for the super quick response!

I gave the option a go, here's the config:

````
sketchybar --add space one left \
  --set one ignore_association=on \
  --set one icon=ﱦ \
  --set one click_script="yabai -m space --focus 1"

sketchybar --add space two left \
  --set two ignore_association=on \
  --set two icon= \
  --set two click_script="yabai -m space --focus 2"

sketchybar --add space three left \
  --set three ignore_association=on \
  --set three icon= \
  --set three click_script="yabai -m space --focus 3"

sketchybar --add space four left \
  --set four ignore_association=on \
  --set four icon=喝 \
  --set four click_script="yabai -m space --focus 4"

sketchybar --add space five left \
  --set five ignore_association=on \
  --set five icon= \
  --set five click_script="yabai -m space --focus 5"

sketchybar --add space six left \
  --set six ignore_association=on \
  --set six icon= \
  --set six click_script="yabai -m space --focus 6"
````

The above displays the 6 icons on the bar but no highlighting happens on any of them. If I add ``associated_space=<space number>`` to each of the above items it's the same as before with the first three will highlight but 4-6 don't. 

It's a small issue and not a big deal, if you have any ideas let me know when you have time.

**Reply by FelixKratz** [reply]
What you might want to try is to explicitly set the `associated_space` as well as the `associated_display` to override the automatic display detection, which in your use case wont work properly as there is no bar on display 2.

**Reply by PhrantiK** [reply]
My original config included both of those options, for example:

````
sketchybar --add space one left \
  --set one associated_space=1 \
  --set one associated_display=1 \
  --set one icon=ﱦ \
  --set one click_script="yabai -m space --focus 1"
````

If you do think of something please let me know but it's no worries if it's not currently possible, the bar is still super rad!

**Comment by PhrantiK** [comment, marked as answer]
I wasn't going to bother you with this as I'm the only one that has asked but about a month ago I managed to brute force a patch that made the space items highlight correctly (IMHO! :)

I just updated it for the recent commit:

https://gist.github.com/PhrantiK/2d88344ec3abd55bd5715dc1cc7d52e7

I'm a monkey with a typewriter so not sure if it's useful but sharing it anyway.

The change to this function: https://github.com/FelixKratz/SketchyBar/blob/fb5923291182e5070b1aef5feaed0caaa2b36074/src/display.c#L213

Came about because for some inexplicable reason the ids & indexes on my monitors swapped randomly when waking from sleep. From this:

```json
[{
        "id":1,
        "uuid":"3AB1D648-6014-498C-92C5-3477DFB6E94C",
        "index":1,
        "label":"",
        "frame":{
                "x":0.0000,
                "y":0.0000,
                "w":3008.0000,
                "h":1692.0000
        },
        "spaces":[1, 2, 3],
        "has-focus":true
},{
        "id":2,
        "uuid":"D7CE0AD5-B301-405A-81BB-8B960CA16D36",
        "index":2,
        "label":"",
        "frame":{
                "x":-1692.0000,
                "y":-539.0000,
                "w":1692.0000,
                "h":3008.0000
        },
        "spaces":[4, 5, 6],
        "has-focus":false
}]
```

to this:

```json
[{
        "id":2,
        "uuid":"3AB1D648-6014-498C-92C5-3477DFB6E94C",
        "index":1,
        "label":"",
        "frame":{
                "x":0.0000,
                "y":0.0000,
                "w":3008.0000,
                "h":1692.0000
        },
        "spaces":[1, 2, 3],
        "has-focus":true
},{
        "id":1,
        "uuid":"D7CE0AD5-B301-405A-81BB-8B960CA16D36",
        "index":2,
        "label":"",
        "frame":{
                "x":-1692.0000,
                "y":-539.0000,
                "w":1692.0000,
                "h":3008.0000
        },
        "spaces":[4, 5, 6],
        "has-focus":false
}]
```

I don't know the greater context for getting the active display ID the original way but nabbing it with `display_active_display_id` seemed to get around it. 

Never had an issue before applying the patch so it's likely (definitely) related to me missing something. 

Thanks again for the rad software dude!
